src/utils/internalLinks.js

```diff
@@ -218,3 +219,14 @@ export const getMessageIdFromLink = (url: string, realm: URL): number => {
 
   return isMessageLink(url, realm) ? parseInt(paths[paths.lastIndexOf('near') + 1], 10) : 0;
 };
+
+export const getStreamTopicURL = (
+  realm: URL,
+  streamId: number,
+  topic: string,
+  streamsById: Map<number, Stream>,
+): URL => {
+  const maybe_get_stream_name = (id: number) => streamsById.get(id)?.name ?? 'unknown';
```

Comment:
The `?? 'unknown'` here is redundant, right? The shared code accepts a nullish result (hence the "maybe" in the name of that callback) and does this fallback. Best to have that in just one place.

Separately: no need to annotate `id: number` -- Flow is able to infer that because of where we pass this callback. If you remove the annotation and then hover on `id`, your editor should still report (with the information the Flow language server gives it) that it's of type `number`. Generally we don't put type annotations on locals like this when they're not required for getting the right type.